### PR TITLE
[CodeCompletion] Implement call pattern completion for subscript

### PR DESCRIPTION
**Pull request by rintaro**
Call signature completion after `expr[`.

rdar://problem/28874899

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by benlangmuir — APPROVED** [review]
LGTM!  Does argument label completion work as well?

File: test/IDE/complete_subscript.swift [hunk 1]
```diff
@@ -49,12 +73,22 @@ func test2<U>(value: MyStruct<U>) {
...
 // INSTANCE_ARCHETYPE: End completions
 
+  let _ = value[#^INSTANCE_ARCHETYPE_BRACKET^#
+// INSTANCE_ARCHETYPE_BRACKET: Begin completions
+// INSTANCE_ARCHETYPE_BRACKET-DAG: Decl[Subscript]/CurrNominal:        ['[']{#(x): Int#}, {#instance: U#}[']'][#Int#];
+// INSTANCE_ARCHETYPE_BRACKET: End completions
+
   let _ = MyStruct<U>[42, #^METATYPE_LABEL^#
 // METATYPE_LABEL: Begin completions, 1 items
 // METATYPE_LABEL-DAG: Keyword/ExprSpecific:               static: [#Argument name#];
```
Comment (rintaro) [inline test/IDE/complete_subscript.swift]: Argument label completion test is here.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test OS X

**Pull request merged**